packages/api/schema/mutation.js

```diff
@@ -327,6 +344,11 @@ export default [
         supply: UpdateProductSupplyInput!
       ): Product
 
+      """
+      Modify plan part of a product
+      """
+      updateProductPlan(productId: ID!, plan: UpdateProductPlanInput!): Product
```

Comment:
You're right, currently only normal PlanProducts can have a plan, but
updateProductWarehousing returns a Product instead of a SimpleProduct and  createProductBundleItem returns a Product instead of a BundleProduct.

So using PlanProduct there would be inconsistent to the other mutations, but maybe we have to refactor that completely to have:

updatePlanProduct
updateConfigurableProduct
updateSimpleProduct
...

Adding the input's dependent on that and return the concrete type

Reply:
I see. Makes sense.